app/main.py

```diff
@@ -1,3 +1,57 @@
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: Union[int, float]) -> None:
+        self.km = km
+
+    def __add__(self, other: Union["Distance", int, float]) -> "Distance":
+        if isinstance(other, int | float):
+            return Distance(self.km + other)
+        return Distance(
+            km=self.km + other.km,
+        )
+
+    def __iadd__(self, other: Union["Distance", int, float]) -> "Distance":
+        if isinstance(other, int | float):
+            self.km += other
+            return self
+        self.km += other.km
+        return self
+
+    def __mul__(self, other: Union[int, float]) -> "Distance":
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: Union[int, float]) -> "Distance":
+        return Distance(round(self.km / other, 2))
+
+    def __lt__(self, other: Union["Distance", int, float]) -> bool:
+        if isinstance(other, int | float):
+            return self.km < other
+        return self.km < other.km
+
+    def __gt__(self, other: Union["Distance", int, float]) -> bool:
+        if isinstance(other, int | float):
+            return self.km > other
+        return self.km > other.km
+
+    def __eq__(self, other: Union["Distance", int, float]) -> bool:
+        if isinstance(other, int | float):
+            return self.km == other
+        return self.km == other.km
+
+    def __le__(self, other: Union["Distance", int, float]) -> bool:
+        if isinstance(other, int | float):
+            return self.km <= other
+        return self.km <= other.km
+
+    def __ge__(self, other: Union["Distance", int, float]) -> bool:
+        if isinstance(other, int | float):
```

Comment:
The use of `int | float` for type checking is incorrect for Python versions prior to 3.10. Use `isinstance(other, (int, float))` instead.